lib/babe/verify.go

```diff
@@ -327,62 +327,58 @@ func (b *verifier) verifyAuthorshipRight(header *types.Header) error {
 		return ErrBadSignature
 	}
 
-	// check if the producer has equivocated, ie. have they produced a conflicting block?
-	// hashes is hashes of all blocks with same block number as header.Number
-	hashes := b.blockState.GetAllBlocksAtDepth(header.ParentHash)
+	equivocated, err := b.verifyBlockEquivocation(header)
+	if err != nil {
+		return fmt.Errorf("could not verify block equivocation for header %s: %w", header.Hash(), err)
+	}
+	if equivocated {
+		return ErrProducerEquivocated
```

Comment:
nit to force using errors.Is(err, ErrGenesisHeader) instead of err == ErrGenesisHeader (as recommended on the Go blog):

```suggestion
		return fmt.Errorf("%w", ErrProducerEquivocated)
```

Also it would be nice to add some context, i.e. the block header hash I guess?